src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -4,45 +4,72 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
-using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Docs.Build
 {
-    internal class GitRepoInfoProvider
+    internal class ContributionInfo
     {
         private static readonly string s_defaultCachePath = Path.Combine(
             Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
             ".docfx",
             "cache",
             "user-profile.json");
 
-        private readonly ConcurrentDictionary<string, GitRepoInfo> _folderRepoInfocache
-            = new ConcurrentDictionary<string, GitRepoInfo>();
+        private readonly ConcurrentDictionary<string, GitRepoInfo> _folderRepoInfocache = new ConcurrentDictionary<string, GitRepoInfo>();
 
-        private readonly Lazy<Dictionary<string, List<GitCommit>>> _fileCommitsCache;
+        private readonly UserProfileCache _userProfileCache;
 
-        private readonly GitUserProfileCache _githubUserProfileCache;
+        private readonly IReadOnlyDictionary<string, List<GitCommit>> _commitsByFile;
 
-        private readonly ReadOnlyDictionary<string, DateTime> _commitTimeBySha;
+        private readonly IReadOnlyDictionary<string, DateTime> _updateTimeByCommit;
 
-        public GitRepoInfoProvider(Docset docset)
+        public ContributionInfo(
```

Comment:
private?

Reply:
nice

---
In reply to: [199423564](https://github.com/dotnet/docfx/pull/2981#discussion_r199423564) [](ancestors = 199423564)